spec/unit/facter/util/fact_rabbitmq_clustername_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+
+RSpec.configure do |config|
+  config.mock_with :rspec
+end
+
+describe Facter::Util::Fact do
+  before { Facter.clear }
+
+  describe 'rabbitmq_clustername' do
+    context 'with value' do
+      before do
+        allow(Facter::Util::Resolution).to receive(:which).with('rabbitmqctl') { true }
+        allow(Facter::Core::Execution).to receive(:execute).with('rabbitmqctl cluster_status 2>&1') { ' {cluster_name,<<"monty">>},' }
+      end
+      it do
+        expect(Facter.fact(:rabbitmq_clustername).value).to eq('monty')
+      end
+    end
+
+    context 'with dashes in hostname' do
+      before do
+        allow(Facter::Util::Resolution).to receive(:which).with('rabbitmqctl') { true }
+        allow(Facter::Core::Execution).to receive(:execute).with('rabbitmqctl cluster_status 2>&1') { ' {cluster_name,<<"monty@rabbit-1">>},' }
+      end
+      it do
+        expect(Facter.fact(:rabbitmq_clustername).value).to eq('monty@rabbit-1')
+      end
+    end
+
+    context 'with dashes in clustername/hostname' do
```

Comment:
Are these from another provider? Are all these valid and expected for cluster names as well as for node names?

Reply:
Since the default of cluster name is actually the hostname of the node which you joined. I would guess all hostname restraints apply.